test_scripts/API/GetWayPoints/common_get_way_points.lua

```diff
@@ -0,0 +1,273 @@
+config.ValidateSchema = false
+
+--[[ Required Shared libraries ]]
+local mobile_session = require("mobile_session")
+local json = require("modules/json")
+
+local commonFunctions = require("user_modules/shared_testcases/commonFunctions")
+local commonSteps = require("user_modules/shared_testcases/commonSteps")
+local commonTestCases = require("user_modules/shared_testcases/commonTestCases")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local hmi_values = require("user_modules/hmi_values")
+
+--[[ Local Variables ]]
+local ptu_table = {}
+local hmiAppIds = {}
+
+local common_get_way_points = {}
+
+common_get_way_points.timeout = 2000
+common_get_way_points.minTimeout = 500
+common_get_way_points.DEFAULT = "Default"
+
+local function checkIfPTSIsSentAsBinary(bin_data)
+  if not (bin_data ~= nil and string.len(bin_data) > 0) then
+    commonFunctions:userPrint(31, "PTS was not sent to Mobile in payload of OnSystemRequest")
+  end
+end
+
+function common_get_way_points.getGetWayPointsConfig()
+  return {
+    keep_context = false,
+    steal_focus = false,
+    priority = "NONE",
+    default_hmi = "NONE",
+    groups = { "Base-4", "GetWayPoints" }
+  }
+end
+
+local function getPTUFromPTS(tbl)
+  tbl.policy_table.consumer_friendly_messages.messages = nil
+  tbl.policy_table.device_data = nil
+  tbl.policy_table.module_meta = nil
+  tbl.policy_table.usage_and_error_counts = nil
+  tbl.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  tbl.policy_table.module_config.preloaded_pt = nil
+  tbl.policy_table.module_config.preloaded_date = nil
+end
+
+local function jsonFileToTable(file_name)
+  local f = io.open(file_name, "r")
+  local content = f:read("*all")
+  f:close()
+  return json.decode(content)
+end
+
+local function ptu(self, ptu_update_func)
+  local function getAppsCount()
+    local count = 0
+    for _, _ in pairs(hmiAppIds) do
+      count = count + 1
+    end
+    return count
+  end
+
+  local policy_file_name = "PolicyTableUpdate"
+  local policy_file_path = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath")
+  local pts_file_name = commonFunctions:read_parameter_from_smart_device_link_ini("PathToSnapshot")
+  local ptu_file_name = os.tmpname()
+  local requestId = self.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+  EXPECT_HMIRESPONSE(requestId)
+  :Do(function()
+      self.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest", { requestType = "PROPRIETARY", fileName = pts_file_name })
+      getPTUFromPTS(ptu_table)
+       local function updatePTU(tbl)
+        tbl.policy_table.functional_groupings["GetWayPoints"] = {
+          rpcs = {
+            GetWayPoints = {
+               hmi_levels = { "BACKGROUND", "FULL", "LIMITED", "NONE" }
+            }
+          }
+        }
+        tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.appID] = common_get_way_points.getGetWayPointsConfig()
```

Comment:
@LitvinenkoIra ptu() function is used in rai_ptu_n() function which is able to register and perform ptu for any app
Currently in line 88 only 1st app will get an updated policies